### DISCUSSION TITLE
Cannot add full amount to piggy bank because canAddAmount returned false

### DISCUSSION THREAD

**Discussion by cowittig** [firefly-iii/firefly-iii#7715] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I cannot add the full amount that is on my account to any piggy bank with message: Error! Could not add €2.50 to "...".
There are older issues with the same problem and marked as fixed, but I am on the latest version (https://github.com/firefly-iii/firefly-iii/issues/6666, https://github.com/firefly-iii/firefly-iii/issues/3152). 

This was working fine for the past year and started to happen a couple of weeks ago. Unfortunately I don't know exactly when.
Log points to floating point issue.

```
2023-07-02 21:51:17       Now executing firefly-iii:report-sum
2023-07-02 21:51:17   [x]  Error: Transactions for user #1 (...) are off by 0.000000000001000959232692!
2023-07-02 21:51:17       Now executing firefly-iii:upgrade-group-information
```
```
[2023-07-02 21:51:32] local.DEBUG: updateStartDate("1M", "2023-07-02")  
[2023-07-02 21:51:32] local.DEBUG: updateEndDate("1M", "2023-07-01")  
[2023-07-02 21:51:39] local.DEBUG: Left on account: 2.499999999998 on 2023-07-02  
[2023-07-02 21:51:39] local.DEBUG: Saved so far: 15.000000000000000000000000  
[2023-07-02 21:51:39] local.DEBUG: Left to save: 15.000000000000  
[2023-07-02 21:51:39] local.DEBUG: Maximum amount: 2.499999999998  
[2023-07-02 21:51:39] local.DEBUG: Compare <= 0? 1, so false  
[2023-07-02 21:51:39] local.ERROR: Cannot add 2.50 because canAddAmount returned false.  
[2023-07-02 21:51:39] local.DEBUG: Now in resetOrder  

```
I am running the docker image.

### Debug information

Debug information generated at 2023-07-02 22:06:38 Europe/Amsterdam for Firefly III version **6.0.16**.

| Scope | Version |
| --- | --- |
| Firefly III | 6.0.16 |
| Firefly III API | 2.0.4 |
| Build | 506, 27-06-2023 13:09:05 UTC |
| Base Build | 10, 21-06-2023 21:55:03 CEST |
| DB version | 19 (exp. 19) |
| PHP | `8.2.7` |
| Host | `Linux` |

| System info | Value |
| --- | --- |
| System TZ | Europe/Amsterdam |
| Browser TZ | Europe/Berlin |
| App environment | local |
| App debug mode | true |
| App cache driver | file |
| App logging | debug, stack |
| Display errors | Off |
| Error reporting | ALL errors |
| Max upload | 67108864 (64 MB) |
| Interface | apache2handler |
| Default language | en_US |
| Default locale | equal |
| BCscale | 12 |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | mysql |
| Trusted proxies (.env) | `` |

| User info | Value |
| --- | --- |
| Login provider | eloquent  |
| User guard | web  |
| Headers | N/A, N/A  |
| Stateful domains |  |
| Session start | 2023-07-01 00:00:00 |
| Session end | 2023-07-31 23:59:59 |
| Session first | 2021-03-28 00:00:00 |
| User ID | 1 |
| User language | en_US |
| User locale | en_US |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 |

**Comment by JC5** [comment, marked as answer]
Create a backup of your database and then run this command. 

`docker exec -it CONTAINER php artisan firefly-iii:force-decimal-size `

**Comment by cowittig** [comment]
Fixed, thank you! 14 transactions were corrected.
